### PR TITLE
fix: avoid global declaration collisions in cjs

### LINKED ISSUE #15604: npm URL throws: `Identifier 'process' has already been declared`
Happens with e.g. `commander`:

```js
import { program } from 'npm:commander';
```

```
Download https://registry.npmjs.org/commander
Download https://registry.npmjs.org/commander/-/commander-9.4.0.tgz
error: Uncaught SyntaxError: Identifier 'process' has already been declared
    at Object.evalContext (deno:core/01_core.js:297:14)
    at wrapSafe (deno:ext/node/02_require.js:692:27)
    at Module._compile (deno:ext/node/02_require.js:703:29)
    at Object.Module._extensions..js (deno:ext/node/02_require.js:740:12)
    at Module.load (deno:ext/node/02_require.js:630:34)
    at Function.Module._load (deno:ext/node/02_require.js:487:14)
    at Module.require (deno:ext/node/02_require.js:652:21)
    at require (deno:ext/node/02_require.js:777:18)
    at Object.<anonymous> (file:///C:/Users/GJZwiers/AppData/Local/deno/npm/registry.npmjs.org/commander/9.4.0/index.js:2:21)
    at Object.<anonymous> (file:///C:/Users/GJZwiers/AppData/Local/deno/npm/registry.npmjs.org/commander/9.4.0/index.js:29:4)
```

```
deno --version

deno 1.25.0 (release, x86_64-pc-windows-msvc)
v8 10.6.194.5
typescript 4.7.4
```

### PR DESCRIPTION
**Pull request by dsherret**
Also adds the ability for us to have custom npm packages in the tests, which is what most of this PR is.

Closes #15604

### CODE REVIEW

File: ext/node/02_require.js [hunk 1]
```diff
@@ -656,11 +656,10 @@
...
 
   Module.wrapper = [
-    // TODO:
-    // We provide non standard timer APIs in the CommonJS wrapper
+    // We provide the non-standard APIs in the CommonJS wrapper
     // to avoid exposing them in global namespace.
-    "(function (exports, require, module, __filename, __dirname, globalThis) { (function (exports, require, module, __filename, __dirname, globalThis, Buffer, clearImmediate, clearInterval, clearTimeout, global, process, setImmediate, setInterval, setTimeout) {",
-    "\n}).call(this, exports, require, module, __filename, __dirname, globalThis, globalThis.Buffer, globalThis.clearImmediate, globalThis.clearInterval, globalThis.clearTimeout, globalThis.global, globalThis.process, globalThis.setImmediate, globalThis.setInterval, globalThis.setTimeout); })",
+    "(function (exports, require, module, __filename, __dirname, globalThis) { const { Buffer, clearImmediate, clearInterval, clearTimeout, global, process, setImmediate, setInterval, setTimeout} = globalThis; (function () {",
+    "\n}).call(this); })",
```
Comment (dsherret) [inline ext/node/02_require.js:662]: Fix is here.

**Comment by dsherret** [comment]
The stack overflows on the Windows CI is because of swc parsing the _pico.mjs module. I'm not sure why it suddenly started happneing, but it happens locally for me too and I increased the stack size to prevent it... https://github.com/denoland/deno/runs/8027046381?check_suite_focus=true

File: test_util/src/npm.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  pub registry_file: String,
+  pub tarballs: HashMap<String, Vec<u8>>,
+}
+
+/// Creates tarballs and a registry json file for npm packages
+/// in the `testdata/npm/registry/@denotest` directory.
+pub struct CustomNpmPackageCache(HashMap<String, CustomNpmPackage>);
+
+impl CustomNpmPackageCache {
+  pub fn load() -> Result<Self, anyhow::Error> {
```
Comment (dsherret) [inline test_util/src/npm.rs:34]: In the future, this should probably be lazy so it's faster. I just wrote this as a first pass.

File: cli/tests/testdata/npm/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# npm test data
+
+This folder contains test data for npm specifiers.
+
+## Registry
+
```
Comment (bartlomieju) [inline cli/tests/testdata/npm/README.md:6]: This is great

**Review by bartlomieju — APPROVED** [review]
LGTM, nice fix!

**Pull request merged**